File: app/src/main/java/com/stupidtree/hichat/ui/search/SearchTrigger.java
package com.stupidtree.hichat.ui.search;

import com.stupidtree.hichat.ui.base.Trigger;

public class SearchTrigger extends Trigger {
    private String searchText;

    public String getSearchText() {
        return searchText;
    }

    public void setSearchText(String searchText) {
        this.searchText = searchText;
    }

    public static SearchTrigger getSearchInstance(String text){
        SearchTrigger st = new SearchTrigger();
        st.setSearchText(text);
        st.setActioning();
        return st;
    }
}

File: app/src/main/java/com/stupidtree/cloudliter/ui/search/SearchTrigger.kt
package com.stupidtree.cloudliter.ui.search

import com.stupidtree.component.data.Trigger

class SearchTrigger : Trigger() {
    var searchText: String? = null

    companion object {
        @JvmStatic
        fun getSearchInstance(text: String?): SearchTrigger {
            val st = SearchTrigger()
            st.searchText = text
            st.setActioning()
            return st
        }
    }
}